tests/test_data_iterator.py

```diff
@@ -39,8 +39,12 @@ def test_csv_full_iterator(chunk_size: Optional[int]) -> None:
         for idx, df in enumerate(dfs):
             assert df.shape[0] == DEFAULT_CHUNK_SIZE_LIST[idx]
     assert dataframe.shape[0] == sum(df_lengths)
-    for dt in {"crawled_at", "harvested_at", "published_at"}:
+    for dt in ["crawled_at", "harvested_at", "published_at"]:
         concat_df[dt] = concat_df[dt].astype("str")
+
+    dataframe = dataframe.sort_values(by="signal_id").reset_index(drop=True)
```

Comment:
why sort?

Reply:
because the main data file we had earlier was sorted by `published_at` and `harvested_at`. Current data is not. That makes the code more flexible.

Reply:
but you're sorting in the test not in the actual code